### ISSUE TITLE
Resident Evil 3 Nemesis Spanish Version Warning Screen Freeze

### ISSUE THREAD

**Issue by JaegerFox** [stenzek/duckstation#2011]
**Game details**
SLES-02532 Resident Evil 3 Nemesis (PAL) Spanish

**Description of the issue / bug**
Game freezes in the Warning Screen

**Controller Troubleshoot Report**
1. No
2. Xinput

**Steps to reproduce**
Just starting the game

**Enhancements information**
with and without enhancements

**Hardware/software**
D3D11/OpenGL

**Emulator version**
Latest Preview 0.1-3850-g9652e3cb (master)

**Additional context**
Im ussing Sbi file

**Comment by stenzek** [comment]
Invalid. Game is running fine here.

Make sure your dump is valid and high risk enhancements are disabled, in particular disc read speedup.

**Comment by JaegerFox** [comment]
I repeat again I am using a valid sbi file as well as a valid image.
The game ran fine before this rolling release # 1878 from then on the game does not load using recompiler, the other two modes work fine.

And it is not necessary that from this problem a change has been made to "prohibit" the execution of a game without an sbi, the message was clear...

**Comment by waspennator** [comment]
I can vouch for him, I tried testing out the version of the game in question with the SBI dump and it still froze on my end too.

**Comment by stenzek** [comment]
Ah, indeed, it is a recompiler issue.. I had it off from an earlier test.

**Comment by stenzek** [comment]
So, this is actually not super straightforward. Currently I implement the cop0/debug breakpoints (necessary for some cheat carts and FreePSXBoot) by switching to the interpreter when they're enabled. But RE3 leaves the breakpoints enabled, which forces it to run with the interpreter when it's not necessary, which means it runs much slower.

I guess I'll have to implement the breakpoint check in the JIT dispatcher. Won't be _too_ difficult to do, but for now I've just gated them behind "Recompiler Memory Exceptions". That way these libcrypt games will run entirely in the recompiler, which is more important than having FreePSXBoot work :P

> And it is not necessary that from this problem a change has been made to "prohibit" the execution of a game without an sbi, the warning message was clear...

I needed to do that anyway, Android users can't read and always click through it. If you _really_ want to run without the sbi, there's an advanced option to force it to boot anyway.
